Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -231,18 +334,25 @@ internal void ClearIdleSessions()
                 {
                     session.close();
                 }
+                _idleSessions.ForEach(session => _waitingQueue.OnResourceDecrease());
                 _idleSessions.Clear();
             }
         }
 
         internal async void ClearAllPoolsAsync()
         {
             s_logger.Debug("SessionPool::ClearAllPoolsAsync");
-            foreach (SFSession session in _idleSessions)
+            IEnumerable<SFSession> idleSessionsCopy;
+            lock (_sessionPoolLock)
+            {
+                idleSessionsCopy = _idleSessions.Select(session => session);
+                _idleSessions.ForEach(session => _waitingQueue.OnResourceDecrease());
```

Comment:
yes, it was something like idle session counter. But actually we don't need to know the exact count for idle sessions. All we need is to know only when a new session is added to idle sessions. So I removed OnResourceDecrease() operation because we don't need it.